pkg/storageincentives/agent.go

```diff
@@ -49,25 +55,30 @@ type Agent struct {
 	monitor        Monitor
 	contract       redistribution.Contract
 	batchExpirer   postagecontract.PostageBatchExpirer
+	stake          staking.RedistributionStatus
 	reserve        postage.Storer
+	nodeState      storage.StateStorer
 	sampler        storage.Sampler
 	overlay        swarm.Address
 	quit           chan struct{}
 	wg             sync.WaitGroup
+	nodeStatus     NodeStatus
+	mtx            sync.Mutex
+	erc20Service   erc20.Service
+	initialBalance *big.Int // current balance of the node before starting the round
+}
+
+// NodeStatus provide internal status of the nodes in the redistribution game
+type NodeStatus struct {
+	State  string   `json:"state"`
```

Comment:
Why is this of type `string` and not `PhaseType`?

Reply:
Good suggestion.